### PR TITLE
Deploy FSX

### PR DESCRIPTION
**Pull request by henriqueribeiro**
*Description of changes:*

- add FSx for Lustre deployment templates
- updated the templates for handling SSD and HDD types of FSx for lustre file systems
- Add FSx security group update to root templete to include Batch Security Group.

Thanks to [Sarthak](https://github.com/spatel-gfb) to help me working on this.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/templates/gwfcore/gwfcore-batch.template.yaml [hunk 1]
```diff
@@ -183,7 +194,7 @@ Resources:
         SecurityGroupIds:
           - !Ref SecurityGroup
         SpotIamFleetRole: !Ref SpotFleetRoleArn
-        Subnets: !Ref SubnetIds
+        Subnets: !If [FsxCondition, !Split [',', !Ref FSxSubnetId], !Ref SubnetIds]
```
[the remark this answers is not in the document]
Comment (spatel-gfb) [reply src/templates/gwfcore/gwfcore-batch.template.yaml:197]: @markjschreiber : This was done as FSxSubnetId is a string and Subnets field needs a list of subnets and hence to convert a string to list, this logic was used.

File: src/templates/gwfcore/gwfcore-fsx.template.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      FileSystemType: "LUSTRE"
+      LustreConfiguration:
+        AutoImportPolicy: "NEW_CHANGED"
+        DeploymentType: "PERSISTENT_1"
+        DriveCacheType: 
+          Fn::If:
+            - TypeCheck
+            - !Ref AWS::NoValue
+            - "NONE"
+        ExportPath: !Sub s3://${S3BucketName}
```
[the remark this answers is not in the document]
Comment (henriqueribeiro) [reply src/templates/gwfcore/gwfcore-fsx.template.yaml:88]: @markjschreiber it makes sense to support multiple buckets and FSxL FS but for now just trying with one.

File: src/templates/gwfcore/gwfcore-batch.template.yaml [hunk 3]
```diff
@@ -87,6 +89,13 @@ Parameters:
   Ec2InstanceProfileArn:
     Type: String
     Description: The IAM Role ARN for the Batch service
+  FSxSubnetId:
+    Type: String
+    Description: 'Subnet you want FSx for Lustre to launch in. We recommend a private subnet. NOTE: Must be from the VPC provided.'
+    Default: "No"
+
+Conditions:
+  FsxCondition: !Not [ !Equals [!Ref FSxSubnetId, "No"]]
```
[the remark this answers is not in the document]
Comment (henriqueribeiro) [reply src/templates/gwfcore/gwfcore-batch.template.yaml:98]: Done!

File: src/templates/gwfcore/gwfcore-fsx.template.yaml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+  VpcId:
+    Type: AWS::EC2::VPC::Id
+    Description: The VPC to create security groups
+  SubnetId:
+    Type: String
+    Description: "Subnet you want your FSx for lustre file system to launch in. Ensure Batch compute environment is also launched in that subnet only."
+  FSxStorageVolumeSize:
+    Type: Number
+    Default: 0
+    Description: The initial size of the FSx volume to be used in GB
```
[the remark this answers is not in the document]
Comment (henriqueribeiro) [reply src/templates/gwfcore/gwfcore-fsx.template.yaml]: Done

**Comment by henriqueribeiro** [comment]
Closing this since https://github.com/aws-samples/aws-genomics-workflows/pull/186 has more updates.

**Pull request closed without merging**